department-of-veterans-affairs/caseflow
# Handle when VSO employee can't access file number

**Why**: The controller design was resulting in a double render error
`AbstractController::DoubleRenderError` in the scenario where a VSO
employee doesn't have access to the file number. This scenario was not
tested.

This PR adds tests for that scenario, and others that weren't tested,
and also cleans up the controller by moving the business logic in
separate classes.

Fixes Sentry errors such as:
https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/3663/events/241077/

Resolves #{github issue number}

### Acceptance Criteria
- When signed in as a VSO employee, searching for a claims file number that you don't have access to doesn't result in a `AbstractController::DoubleRenderError` server error.

### Testing Plan
1. Sign in as a VSO employee
2. Search for a claims file number that you don't have access to
Expected Result: you see a helpful error message that says "You don't have access to this claims file number"
